discovery-provider/src/tasks/index_trending.py

```diff
@@ -238,9 +349,36 @@ def floor_time(dt: datetime, interval_seconds: int):
     return dt + timedelta(0, rounding - seconds, -dt.microsecond)
 
 
+def find_min_block_above_timestamp(block_number: int, min_timestamp: datetime, web3):
+    """
+    finds the minimum block number above a timestamp
+    This is needed to ensure consistency across discovery nodes on the timestamp/blocknumber
+    of a notification for updates to trending track
+    returns a tuple of the blocknumber and timestamp
+    """
+    curr_block_number = block_number
+    block = get_block(web3, block_number)
+    while datetime.fromtimestamp(block["timestamp"]) > min_timestamp:
```

Comment:
Is there a maximum amount of checks we want to perform in case this thing loops for a really long time?